nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+  networks: string[]; // Array of network names the wallet provider has available for the dapp developer to connect to.
+  defaultNetwork: string; // Network the wallet is currently set to.
+}
+</pre>
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getNetworks()
+.then(response => {
+  const {
+    networks,
+    defaultNetwork,
+  } = response.networks;
+
+  console.log('Networks: ' + networks);
+  // eg. ["MainNet", "TestNet", "PrivateNet"]
+
+  console.log('Default network: ' + defaultNetwork);
+  // eg. "MainNet"
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+- The list of NEO networks that the wallet is able to connect to. The return values will be used by the dapp developer to communicate to which network they would like their request to be directed to. The wallet provider will be responsible for the logistics of which node to submit any request to for each network alias provided.
+- The wallet provider is to also return the default network that the wallet UI is currently set to by user.
+
+Example
+<pre>
+{
+  networks: ["MainNet", "TestNet", "PrivateNet"],
+  defaultNetwork: "TestNet",
+}
+</pre>
+
+
+====getAccount====
+Return the Account that is currently connected to the dApp.
+
+
+'''Method Interface'''
+
+<pre>
+function getAccount(): Promise<{ address: string, label: string }>
+</pre>
+
+
+'''Input arguments'''
+
+None
+
+
+'''Success return value'''
+
+<pre>
+interface Account {
+  address: string; // Address of the connected account
+  label?: string; // A label the users has set to identify their wallet
+}
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getAccount()
+.then((account: Account) => {
+  const {
+    address,
+    label
+  } = account;
+
+  console.log('Provider address: ' + address);
+  console.log('Provider account label (Optional): ' + label);
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+- The address and label of account currently selected by the user in the wallet interface
+
+Example
+<pre>
+{
+  address: 'AeysVbKWiLSuSDhg7DTzUdDyYYKfgjojru',
```

Comment:
Ok, looks like Address is indeed Base58 string format (better inform this).

Reply:
added